compiler-rt/test/asan/TestCases/Linux/dlopen-mixed-c-cxx.c

```diff
@@ -1,7 +1,10 @@
 // RUN: %clangxx_asan -xc++ -shared -fPIC -o %t.so - < %s
 // RUN: %clang_asan %s -o %t.out -ldl
 //
-// RUN: { env ASAN_OPTIONS=verbosity=1 %t.out %t.so || : ; } 2>&1 | FileCheck %s
+// The program can potentially return a non-zero exit code, so use || : to
+// ensure command returns true
+// RUN: env ASAN_OPTIONS=verbosity=1 %t.out %t.so &> %t_env || :
```

Comment:
The initial version of this commit used `not`: https://github.com/llvm/llvm-project/commit/cbd28cd05bb11e9d76d71c6cc2d38d89dbb1fe1a but for some reason that was changed in https://github.com/llvm/llvm-project/commit/a30a4a35ecbd2ebff79de8a193e7045b17da3ff5 

Maybe removing `not` was used to work around the missing `UNSUPPORTED:` that was only added in https://github.com/llvm/llvm-project/commit/307daa71a8f0bb258d4c89d5bad2579f468162ca CC: @serge-sans-paille @hahnjo 

I'd be tempted to restore this to using `not` instead of `|| :` and see if the builders are happy.

Reply:
I replaced `||:` with `not` in my most recent commit. The tests are passing when I run things locally - still need to wait for builders though.